easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,189 @@
+
+
+class EB_MXNet(MakeCp):
+    """Easyblock to build and install MXNet"""
+
+    @staticmethod
+    def extra_options():
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def __init__(self, *args, **kwargs):
+        """Initialize custom class variables."""
+        super(EB_MXNet, self).__init__(*args, **kwargs)
+
+        self.mxnet_src_dir = None
+        self.py_ext = PythonPackage(self, {'name': 'mxnet'})
+        self.py_ext.module_generator = self.module_generator
+        self.r_ext = RPackage(self, {'name': 'mxnet'})
+        self.r_ext.module_generator = self.module_generator
+
+    def extract_step(self):
+        """
+        Prepare a combined MXNet source tree. Move all submodules
+        to their right place.
+        """
+        # Extract everything into separate directories.
+        super(EB_MXNet, self).extract_step()
+
+        # Find the full path to the directory that was unpacked from mxnet-*.tar.gz.
+        for srcdir in os.listdir(self.builddir):
+            if srcdir.startswith("mxnet-"):
+                self.mxnet_src_dir = os.path.join(self.builddir, srcdir)
+                break
+
+        if self.mxnet_src_dir is None:
+            raise EasyBuildError("Could not find the MXNet source directory")
+
+        self.log.debug("MXNet dir is: %s", self.mxnet_src_dir)
+
+        for srcdir in os.listdir(self.builddir):
+            if srcdir.startswith("mxnet-"):
+                continue
+            else:
+                submodule, _, _ = srcdir.rpartition('-')
+                try:
+                    newdir = os.path.join(self.mxnet_src_dir, submodule)
+                    olddir = os.path.join(self.builddir, srcdir)
+                    # first remove empty existing directory
+                    os.rmdir(newdir)
+                    shutil.move(olddir, newdir)
+                except IOError, err:
+                    raise EasyBuildError("Failed to move %s to %s: %s", olddir, newdir, err)
+
+        # the nnvm submodules has dmlc-core as a submodule too. Let's up a symlink in place.
+        try:
+            newdir = os.path.join(self.mxnet_src_dir, "nnvm", "dmlc-core")
+            olddir = os.path.join(self.mxnet_src_dir, "dmlc-core")
+            os.rmdir(newdir)
+            symlink(olddir, newdir)
+        except IOError, err:
+            raise EasyBuildError("Failed to symlink %s to %s: %s", olddir, newdir, err)
+
+    def configure_step(self):
+        """Patch 'config.mk' file to use EB stuff"""
+        copy_file('make/config.mk', '.')
+
+        regex_subs = [
+            (r"export CC = gcc", r"# \g<0>"),
+            (r"export CXX = g\+\+", r"# \g<0>"),
+            (r"(?P<var>ADD_LDFLAGS\s*=)\s*$", r"\g<var> %s" % os.environ['EBVARCFLAGS']),
+            (r"(?P<var>ADD_CFLAGS\s*=)\s*$", r"\g<var> %s" % os.environ['EBVARLDFLAGS']),
```

Comment:
you have these mixed up?

also, don't use `EBVAR`, cfr. https://github.com/hpcugent/easybuild-framework/issues/1219